### PR TITLE
fixes #154

### LINKED ISSUE #154: Installer continues if user presses cancel on interface or protocol selection screens
If the user really wanted to cancel, they can hit Ctrl+C after it continues on, but it would be nice for the cancel button to actually work.

### PR DESCRIPTION
**Pull request by jacobsalmela**
Lets the user stop the install if cancel is selected.

### CODE REVIEW

**Comment by PromoFaux** [comment]
Does the new code [here](https://github.com/jacobsalmela/pi-hole/blob/332baed4bb2fa99546e3a0833bbfff62aecd28bf/automated%20install/basic-install.sh#L151) make [this bit](https://github.com/jacobsalmela/pi-hole/blob/332baed4bb2fa99546e3a0833bbfff62aecd28bf/automated%20install/basic-install.sh#L174) redundant?

**Comment by jacobsalmela** [comment]
I don't think so.  The first code you linked to exits if the user hits the cancel button.  The other exits if there are no protocols selected.  I'll have to test it out, but that's how I read it.

**Comment by dschaper** [comment]
I think the current state of development is ready for master, we've tested all the currently committed code with @gh0stshell last night and all went well. Any objections/confirmations for merge?

**Comment by jacobsalmela** [comment]
I feel good about it.

**Comment by dschaper** [comment]
Merged development into master and tagged with v2.3. (I promise I'm working on the packaging, I'm trying to set it up so that the packaging is ready for more than just Raspbian, in light of some of the comments on reddit regarding the platforms folks are running under.)

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -338,12 +349,12 @@ $SUDO echo "::: ...done."
...
 $SUDO echo "::: Downloading and installing pihole custom index page..."
-if [ -d "/var/www/html/pihole" ]; then	
+if [ -d "/var/www/html/pihole" ]; then
   $SUDO echo "::: Existing page detected, not overwriting"
-else  
+else
 	$SUDO mkdir /var/www/html/pihole
 	$SUDO mv /var/www/html/index.lighttpd.html /var/www/html/index.lighttpd.orig
-	$SUDO curl -o /var/www/html/pihole/index.html https://raw.githubusercontent.com/jacobsalmela/pi-hole/master/advanced/index.html	
+	$SUDO curl -o /var/www/html/pihole/index.html https://raw.githubusercontent.com/jacobsalmela/pi-hole/master/advanced/index.html
```
Comment (dschaper) [inline automated install/basic-install.sh]: Just need to change this line for the new repo.

**Comment by dschaper** [comment]
Seems to exit every time I run it:
```
+ tmpLog=/tmp/pihole-install.log
+ instalLogLoc=/etc/pihole/install.log
++ tput lines
+ rows=47
++ tput cols
+ columns=117
+ r=23
+ c=58
++ ip route get 8.8.8.8
++ awk '{for(i=1;i<=NF;i++)if($i~/dev/)print $(i+1)}'
+ IPv4dev=eth1
++ ip -o -f inet addr show dev eth1
++ awk '{print $4}'
++ awk 'END {print}'
+ IPv4addr=192.168.1.182/24
++ ip route get 8.8.8.8
++ awk '{print $3}'
+ IPv4gw=192.168.1.1
++ ip -o link
++ awk '{print $2}'
++ cut -d: -f1
++ grep -v lo
+ availableInterfaces='eth0
eth1'
+ dhcpcdFile=/etc/dhcpcd.conf
+ welcomeDialogs
+ whiptail --msgbox --backtitle Welcome --title 'Pi-hole automated installer' 'This installer will transform your Raspberry Pi into a network-wide ad blocker!' 23 58
+ whiptail --msgbox --backtitle Plea --title 'Free and open source' 'The Pi-hole is free, but powered by your donations:  http://pi-hole.net/donate' 23 58
+ whiptail --msgbox --backtitle 'Initating network interface' --title 'Static IP Needed' 'The Pi-hole is a SERVER so it needs a STATIC IP ADDRESS to function properly.

In the next section, you can choose to use your current network settings (DHCP) or to manually edit them.' 23 58
+ backupLegacyPihole
+ [[ -f /etc/dnsmasq.d/adList.conf ]]
+ :
+ chooseInterface
+ [[ -n '' ]]
+ echo 'Cancelling installation.'
Cancelling installation.
+ exit
```

**Comment by PromoFaux** [comment]
Hmm, looks like its falling into the cancel check.

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -103,19 +103,24 @@ In the next section, you can choose to use your current network settings (DHCP)
 chooseInterface()
 {
 # Turn the available interfaces into an array so it can be used with a whiptail dialog
-interfacesArray=()
-firstloop=1
+if [[ $chooseInterfaceOptions ]];then
```
Comment (dschaper) [inline automated install/basic-install.sh]: I think this is where it's failing, that variable is coming back as null.

Comment (jacobsalmela) [inline automated install/basic-install.sh]: Hmmm.  I'll look at it again.

**Comment by PromoFaux** [comment]
Fixed in 29e0e17

**Pull request closed without merging**